### DISCUSSION TITLE
Editing Words and Expressions for projects in diferent languages

### DISCUSSION THREAD

**Discussion by GuilhermeHirsch** [Budibase/budibase#2023] [Ideas]
Hey guys! 

I have a feature suggestion that is important to people who have non-English-language businesses and projects.
A setup screen that allows you to change words and expressions that are not editable. 

For example:
- The months and days of the datepicker;
- The messages that appear when a row is saved, updated...
- User login screen expressions;
- The word "page" from the pagination.
- And everything else that appears to the end user that is not editable.

For me this is a differential in the development tools I use.

It could be something very simple that the appsheet uses:
![Sem título](https://user-images.githubusercontent.com/33108596/126178861-f05674fd-5fc7-4292-8fe4-67924fd58a81.jpg)

**Comment by mjashanks** [comment]
Agreed - this is definitely something that we will need to do at some point. Thanks!

**Reply by GuilhermeHirsch** [reply]
I wonder how busy you are with other aspects of the application. But I have already registered this wish here. Thanks for answering, @mjashanks!

**Reply by mjashanks** [reply]
For reference, this is also mentioned here: https://github.com/Budibase/budibase/discussions/1150.

@GuilhermeHirsch just to clarify your needs - are you wanting to build an app into one language? or multiple?

**Reply by GuilhermeHirsch** [reply]
An app with only one language, @mjashanks.

**Reply by GuilhermeHirsch** [reply]
I see this functionality with great eyes, because, for example, there are many people here in Brazil developing solutions with low code tools. And so far, almost none of them have this language support. Glad to hear that you have this interest.

**Reply by nunopoco** [reply]
Any updates on UI and components translation?
I've started using Budibase but now I see a lot of the components have this issue and we can't translate them. The date picker is one where the experience gets worse as the months are impossible to understand for a non-english reader. 
Is there a workaround on how to do this?

**Reply by alcompstudio** [reply]
For me, this is also very important. There is a lack of multilingual functionality, both for the platform itself (so that non-English-speaking developers can comfortably work on creating web applications), and for external user interfaces.

I have seen that in some other no-code platforms this is implemented using aliases (label substitution). I'm not a programmer, I don't know how it should work...

**Reply by dealanis** [reply]
We are currently deploying budibase for internal use. Im okay if the builder is only in english, but as others says, the components have some functions that are impossible to translate. like: tables, date picker, user porfile, etc.

Would love to see this function soon, my team only knows spanish and the language barrier can be an issue. I understant that there are AI tools right now that can do this automatically.

All the texts that I can control are already in spanish. Thanks!
